SMTP_handler.py

```diff
@@ -0,0 +1,520 @@
+        # add alias name from domain
+        if alias.custom_domain.name:
+            LOG.d(
+                "Put domain default alias name %s in from header",
+                alias.custom_domain.name,
+            )
+            from_header = formataddr((alias.custom_domain.name, alias.email))
+
+    LOG.d("From header is %s", from_header)
+    add_or_replace_header(msg, headers.FROM, from_header)
+
+    replace_original_message_id(alias, email_log, msg)
+
+    if not msg[headers.DATE]:
+        date_header = formatdate()
+        LOG.w("missing date header, add one")
+        msg[headers.DATE] = date_header
+
+    msg[headers.SL_DIRECTION] = "Reply"
+    msg[headers.SL_EMAIL_LOG_ID] = str(email_log.id)
+
+    LOG.d(
+        "send email from %s to %s, mail_options:%s,rcpt_options:%s",
+        alias.email,
+        contact.website_email,
+        envelope.mail_options,
+        envelope.rcpt_options,
+    )
+
+    if should_add_dkim_signature(alias_domain):
+        add_dkim_signature(msg, alias_domain)
+
+    # generate a mail_from for VERP <- Can be kept for SMTP as well
+    verp_mail_from = f"{BOUNCE_PREFIX_FOR_REPLY_PHASE}+{email_log.id}+@{alias_domain}"
+
+    try:
+        sl_sendmail(
+            verp_mail_from,
+            contact.website_email,
+            msg,
+            envelope.mail_options,
+            envelope.rcpt_options,
+            is_forward=False,
+        )
+    except Exception:
+        LOG.w("Cannot send email from %s to %s", alias, contact)
+        EmailLog.delete(email_log.id, commit=True)
+        send_email(
+            mailbox.email,
+            f"Email cannot be sent to {contact.email} from {alias.email}",
+            render(
+                "transactional/reply-error.txt.jinja2",
+                user=user,
+                alias=alias,
+                contact=contact,
+                contact_domain=get_email_domain_part(contact.email),
+            ),
+            render(
+                "transactional/reply-error.html",
+                user=user,
+                alias=alias,
+                contact=contact,
+                contact_domain=get_email_domain_part(contact.email),
+            ),
+        )
+
+    # return 250 even if error as user is already informed of the incident and can retry sending the email
+    return True, status.E200
+
+
+class SMTPAuthenticator:
+    def fail_nothandled(self, message=None) -> AuthResult:
+        return AuthResult(success=False, handled=False, message=message)
+
+    def __call__(self, server, session, envelope, mechanism, auth_data):
+        if mechanism not in ("LOGIN", "PLAIN"):
+            LOG.e("mechanism %s not supported.", mechanism)
+            return self.fail_nothandled("550 Mechanism not supported")
+
+        if not isinstance(auth_data, LoginPassword):
+            LOG.e("Incorrect Format for Credentials")
+            return self.fail_nothandled(status.E501)
+
+        username = (auth_data.login).decode("utf-8")
```

Comment:
we can remove the parenthesis around `auth_data.login` right?

Reply:
I have removed Unnecessary parenthesis.